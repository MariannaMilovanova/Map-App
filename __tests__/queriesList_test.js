import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import QueriesList from '../src/javascripts/components/queriesList/queriesList';

const items = [{id: "59f1bb3e351e3d0c4b8a1d8f", term: "bar", center: {lat: 35.6895, lng: 139.69169999999997}, radius: 3660, date: new Date()},
  {id: "59f1bb3e351e3d0c4b8a1d82", term: "cinema", center: {lat: 35.6895, lng: 139.69169999999997}, radius: 3660, date: new Date()}];

it('should match its snapshot with query items', () => {
  const list = toJson(mount(<QueriesList searchHistory={items} />));

  expect(list).toMatchSnapshot();
});

it('img div should have onClick function', () => {
    const component = mount(<QueriesList searchHistory={[items[0]]} deleteQuery={() => {}}/>);
    component.find('img').simulate('click', { target: {
      value: 'Click function' }
    });

    expect(toJson(component)).toMatchSnapshot();
});