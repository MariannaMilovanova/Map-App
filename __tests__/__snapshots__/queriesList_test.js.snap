// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`img div should have onClick function 1`] = `
<QueriesList
  deleteQuery={[Function]}
  searchHistory={
    Array [
      Object {
        "center": Object {
          "lat": 35.6895,
          "lng": 139.69169999999997,
        },
        "date": 2017-10-26T14:38:16.390Z,
        "id": "59f1bb3e351e3d0c4b8a1d8f",
        "radius": 3660,
        "term": "bar",
      },
    ]
  }
>
  <div
    className="queries-list-wrapper"
  >
    <div
      className="search-history-table"
    >
      <div>
        <div
          className="search-history-item"
          key="0"
        >
          <div
            className="search-history-cell-img"
          >
            <img
              className="recycle-bin-img"
              onClick={[Function]}
              src="test-file-stub"
            />
            <div>
              bar
            </div>
          </div>
          <div
            className="search-history-cell"
          >
            35.6895
          </div>
          <div
            className="search-history-cell"
          >
            139.6917
          </div>
          <div
            className="search-history-cell-radius"
          >
            4
            km
          </div>
          <div
            className="search-history-cell-date"
          >
            Oct 26, 2017 05:38 PM
          </div>
        </div>
      </div>
    </div>
  </div>
</QueriesList>
`;

exports[`should match its snapshot with query items 1`] = `
<QueriesList
  searchHistory={
    Array [
      Object {
        "center": Object {
          "lat": 35.6895,
          "lng": 139.69169999999997,
        },
        "date": 2017-10-26T14:38:16.390Z,
        "id": "59f1bb3e351e3d0c4b8a1d8f",
        "radius": 3660,
        "term": "bar",
      },
      Object {
        "center": Object {
          "lat": 35.6895,
          "lng": 139.69169999999997,
        },
        "date": 2017-10-26T14:38:16.390Z,
        "id": "59f1bb3e351e3d0c4b8a1d82",
        "radius": 3660,
        "term": "cinema",
      },
    ]
  }
>
  <div
    className="queries-list-wrapper"
  >
    <div
      className="search-history-table"
    >
      <div>
        <div
          className="search-history-item"
          key="0"
        >
          <div
            className="search-history-cell-img"
          >
            <img
              className="recycle-bin-img"
              onClick={[Function]}
              src="test-file-stub"
            />
            <div>
              bar
            </div>
          </div>
          <div
            className="search-history-cell"
          >
            35.6895
          </div>
          <div
            className="search-history-cell"
          >
            139.6917
          </div>
          <div
            className="search-history-cell-radius"
          >
            4
            km
          </div>
          <div
            className="search-history-cell-date"
          >
            Oct 26, 2017 05:38 PM
          </div>
        </div>
        <div
          className="search-history-item"
          key="1"
        >
          <div
            className="search-history-cell-img"
          >
            <img
              className="recycle-bin-img"
              onClick={[Function]}
              src="test-file-stub"
            />
            <div>
              cinema
            </div>
          </div>
          <div
            className="search-history-cell"
          >
            35.6895
          </div>
          <div
            className="search-history-cell"
          >
            139.6917
          </div>
          <div
            className="search-history-cell-radius"
          >
            4
            km
          </div>
          <div
            className="search-history-cell-date"
          >
            Oct 26, 2017 05:38 PM
          </div>
        </div>
      </div>
    </div>
  </div>
</QueriesList>
`;
